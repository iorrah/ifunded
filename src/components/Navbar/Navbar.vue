<template>
  <nav class="navbar container-fluid">
    <div class="flex__space-between">
      <div class="navbar__left">
        <a href="#" class="navbar__link">
          <img src="../../assets/ifunded-logo.svg" class="navbar__logo" />
        </a>

        <ul class="navbar__menu">
          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link">Dashboard</a>
          </li>

          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link">Customers</a>
          </li>

          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link">Developers</a>
          </li>

          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link">Projects</a>
          </li>

          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link navbar__menu-link--active">
              Users
              <i class="navbar__menu-indicator--active"></i>
            </a>
          </li>

          <li class="navbar__menu-item">
            <a href="#" class="navbar__menu-link">Affiliates</a>
          </li>
        </ul>
      </div>

      <div class="navbar__right">
        <div class="navbar__form">
          <div class="material-form">
            <div class="material-form__input-text">
              <div class="group">
                <input
                  data-message="The format of your email is incorrect"
                  type="text"
                  name="_username"
                  placeholder=""
                  required=""
                  class="inpField"
                  id="username"
                  value=""
                />

                <img
                  src="../../assets/search-icon.svg"
                  class="material-form__search-icon"
                />

                <span class="highlight"></span>
                <span class="bar"></span>

                <label>
                  Search
                </label>

                <span class="error-message">
                  Error
                </span>
              </div>
            </div>

            <div class="material-form__input-select">
              <div class="group select-group">
                <div class="jq-selectbox jqselect inpField dropup opened">
                  <select required="required" class="inpField inpField__select">
                    <option label="English" selected="selected"></option>
                    <option label="German" selected=""></option>
                  </select>

                  <div class="jq-selectbox__select">
                    <div class="jq-selectbox__select-text placeholder">
                      Language
                    </div>

                    <div class="jq-selectbox__trigger">
                      <div class="jq-selectbox__trigger-arrow">
                        <img
                          src="../../assets/select-arrows.svg"
                          class="jq-selectbox__select-arrow"
                        />
                      </div>
                    </div>
                  </div>
                </div>

                <span class="highlight"></span>
                <span class="bar"></span>
              </div>
            </div>
          </div>
        </div>

        <a href="#" class="navbar__profile">
          <span class="navbar__salutation">Janet Weaver</span>
          <img src="../../assets/user-avatar.png" class="navbar__avatar">
        </a>
      </div>
    </div>
  </nav>
</template>

<script>
export default {
  name: 'Navbar',
};
</script>
