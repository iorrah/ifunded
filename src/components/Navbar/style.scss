.navbar {
  display: block;
  width: 100%;
  padding-top: 1rem;
  padding-bottom: 1rem;
  height: 4.1rem;
  background: $white;
}

.navbar__link {
  display: inline-block;
  width: 5rem;
  margin: 0 3rem 0 0;
}

.navbar__logo {
  display: block;
  width: 100%;
}

.navbar__menu {
  display: inline-block;
  position: relative;
  top: -0.6rem;
  margin: 0;
}

.navbar__menu-item {
  list-style: none;
  display: inline-block;
  margin: 0 1.3rem 0;
}

.navbar__menu-link {
  color: $black;
  padding: 1.5rem 0;
  font-size: 0.9rem;
  font-weight: 600;
}

.navbar__menu-link--active {
  color: $blue
}

.navbar__menu-indicator--active {
  background: $blue;
  display: block;
  height: 0.2rem;
  width: 100%;
  position: relative;
  bottom: -1.3rem;
  border-radius: 1rem 1rem 0 0;
}

.navbar__form {
  display: inline-block;
  position: relative;
  top: -1.5rem;
  margin: 0 2.5rem 0 0;
}

.navbar__profile {
  display: inline-block;
  width: auto;
  position: relative;
  top: -1.8rem;
}

.navbar__salutation {
  font-size: 0.9rem;
  color: $blue;
  margin: 0 0.7rem 0 0;
  position: relative;
  top: 0.2rem;
}

.navbar__avatar {
  width: 2.5rem;
  vertical-align: middle;
}