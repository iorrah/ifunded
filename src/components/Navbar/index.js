// eslint-disable-next-line import/newline-after-import
import Navbar from './Navbar.vue';
export default Navbar;
