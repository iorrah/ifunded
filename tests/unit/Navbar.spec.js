import { mount } from '@vue/test-utils';
import Navbar from '../../src/components/Navbar/index';

describe('Navbar', () => {
  test('renders correctly', () => {
    const wrapper = mount(Navbar);
    expect(wrapper.element).toMatchSnapshot();
  });
});
