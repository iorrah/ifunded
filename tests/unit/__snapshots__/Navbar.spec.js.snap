// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  class="navbar container-fluid"
>
  <div
    class="flex__space-between"
  >
    <div
      class="navbar__left"
    >
      <a
        class="navbar__link"
        href="#"
      >
        <img
          class="navbar__logo"
          src="../../assets/ifunded-logo.svg"
        />
      </a>
       
      <ul
        class="navbar__menu"
      >
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link"
            href="#"
          >
            Dashboard
          </a>
        </li>
         
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link"
            href="#"
          >
            Customers
          </a>
        </li>
         
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link"
            href="#"
          >
            Developers
          </a>
        </li>
         
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link"
            href="#"
          >
            Projects
          </a>
        </li>
         
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link navbar__menu-link--active"
            href="#"
          >
            
            Users
            
            <i
              class="navbar__menu-indicator--active"
            />
          </a>
        </li>
         
        <li
          class="navbar__menu-item"
        >
          <a
            class="navbar__menu-link"
            href="#"
          >
            Affiliates
          </a>
        </li>
      </ul>
    </div>
     
    <div
      class="navbar__right"
    >
      <div
        class="navbar__form"
      >
        <div
          class="material-form"
        >
          <div
            class="material-form__input-text"
          >
            <div
              class="group"
            >
              <input
                class="inpField"
                data-message="The format of your email is incorrect"
                placeholder=""
                required="required"
                type="text"
                value=""
              />
               
              <img
                class="material-form__search-icon"
                src="../../assets/search-icon.svg"
              />
               
              <span
                class="highlight"
              />
               
              <span
                class="bar"
              />
               
              <label>
                
                Search
              
              </label>
               
              <span
                class="error-message"
              >
                
                Error
              
              </span>
            </div>
          </div>
           
          <div
            class="material-form__input-select"
          >
            <div
              class="group select-group"
            >
              <div
                class="jq-selectbox jqselect inpField dropup opened"
              >
                <select
                  class="inpField inpField__select"
                  required="required"
                >
                  <option
                    label="English"
                    selected="selected"
                  />
                   
                  <option
                    label="German"
                  />
                </select>
                 
                <div
                  class="jq-selectbox__select"
                >
                  <div
                    class="jq-selectbox__select-text placeholder"
                  >
                    
                    Language
                  
                  </div>
                   
                  <div
                    class="jq-selectbox__trigger"
                  >
                    <div
                      class="jq-selectbox__trigger-arrow"
                    >
                      <img
                        class="jq-selectbox__select-arrow"
                        src="../../assets/select-arrows.svg"
                      />
                    </div>
                  </div>
                </div>
              </div>
               
              <span
                class="highlight"
              />
               
              <span
                class="bar"
              />
            </div>
          </div>
        </div>
      </div>
       
      <a
        class="navbar__profile"
        href="#"
      >
        <span
          class="navbar__salutation"
        >
          Janet Weaver
        </span>
         
        <img
          class="navbar__avatar"
          src="../../assets/user-avatar.png"
        />
      </a>
    </div>
  </div>
</nav>
`;
